File: app/src/main/java/com/sxshi/retrofit/utils/FragmentTransactionUtil.java
package com.sxshi.retrofit.utils;

import androidx.annotation.IdRes;
import androidx.annotation.NonNull;
import androidx.fragment.app.Fragment;
import androidx.fragment.app.FragmentManager;
import androidx.fragment.app.FragmentTransaction;

public final class FragmentTransactionUtil {

    private FragmentTransactionUtil() {
        throw new AssertionError();
    }

    public static void addFragment(FragmentManager fragmentManager,
                                   @NonNull Fragment fragment,
                                   @IdRes int fragmentContainerId,
                                   boolean addToBackStack) {
        FragmentTransaction transaction = fragmentManager.beginTransaction();
        transaction.add(fragmentContainerId, fragment);
        if (addToBackStack) {
            transaction.addToBackStack(fragment.getClass().getSimpleName());
        }
        transaction.commit();
    }

    public static void replaceFragment(FragmentManager fragmentManager,
                                       @NonNull Fragment fragment,
                                       @IdRes int fragmentContainerId,
                                       boolean addToBackStack) {
        FragmentTransaction transaction = fragmentManager.beginTransaction();
//    transaction.setCustomAnimations(
//        R.anim.slide_in_bottom, R.anim.zoom_out,
//        R.anim.zoom_out, R.anim.slide_out_bottom);
        transaction.replace(fragmentContainerId, fragment);
        if (addToBackStack) {
            transaction.addToBackStack(fragment.getClass().getSimpleName());
        }
        transaction.commit();
    }
}

File: app/src/main/java/com/sxshi/retrofit/utils/FragmentTransactionUtil.kt
package com.sxshi.retrofit.utils

import androidx.annotation.IdRes
import androidx.fragment.app.Fragment
import androidx.fragment.app.FragmentManager

class FragmentTransactionUtil private constructor() {
    companion object {
        @JvmStatic
        fun addFragment(fragmentManager: FragmentManager,
                        fragment: Fragment,
                        @IdRes fragmentContainerId: Int,
                        addToBackStack: Boolean) {
            val transaction = fragmentManager.beginTransaction()
            transaction.add(fragmentContainerId, fragment)
            if (addToBackStack) {
                transaction.addToBackStack(fragment.javaClass.simpleName)
            }
            transaction.commit()
        }

        fun replaceFragment(fragmentManager: FragmentManager,
                            fragment: Fragment,
                            @IdRes fragmentContainerId: Int,
                            addToBackStack: Boolean) {
            val transaction = fragmentManager.beginTransaction()
            //    transaction.setCustomAnimations(
//        R.anim.slide_in_bottom, R.anim.zoom_out,
//        R.anim.zoom_out, R.anim.slide_out_bottom);
            transaction.replace(fragmentContainerId, fragment)
            if (addToBackStack) {
                transaction.addToBackStack(fragment.javaClass.simpleName)
            }
            transaction.commit()
        }
    }

    init {
        throw AssertionError()
    }
}